README.md

```diff
@@ -264,3 +287,77 @@ build:
 
 You can also override the build command (`make build` by default) with the
 `--build-command` flag.
+
+## Working without a local repository
+
+Monobuild can work without a local clone and checkout of your repository. All of
+the use can be supported with GitHub API calls.
+
+This is an **optimisation** and may not be necessary on relatively small monorepos or
+if your CI caches the repo and only pulls changes since last build.
+
+### Load dependency graph from a file
+
+It is possible to keep the dependencies in a central dependency graph file, which
+is useful for big repos, when you want to work without a full local checkout (e.g. on CI).
+
+Monobuild supports this by allowing a map file to be specified with the `-f` flag.
+For example:
+
+```sh
+$ curl -O https://raw.githubusercontent.com/myorg/myrepo/blob/master/dependencies.monobuild
+$ mononobuild print -f ./dependencies.monobuild --dependencies
+```
+
+The file can be created with the `print` command itself, which will collect the
+manifests and produce the full map.
+
+```sh
+$ monobuild print --full
+app4:
+libs/lib1: libs/lib3
+libs/lib2: libs/lib3
+libs/lib3:
+stack1: !app1, !app2, !app3
+app1: libs/lib1, libs/lib2
+app2: libs/lib2, libs/lib3
+app3: libs/lib3
+```
+
+You can update the dependency map as follows (with a local checkout):
+
+```sh
+$ monobuild print --full > dependencies.monobuild
+```
+
+### Read changed files from standard input
+
+Similarly, the changed files for `diff` can be supplied externally, from
+standard input, e.g.
+
+```
+$ git diff --no-commit-id --name-only -r $(git merge-base master HEAD) | monobuild diff -
+```
+
+**note the hyphen at the end**, indicating the diff should be taken from stdin,
```

Comment:
Could it be a more explicit flag ? like -i and --standard-input Or maybe it's a convention (In that case I'm sorry I didn't know about it)

http://www.informit.com/articles/article.aspx?p=2854374&seqNum=5

Reply:
I took that from Docker's convention for `docker build` https://docs.docker.com/engine/reference/commandline/build/. 

> Instead of specifying a context, you can pass a single Dockerfile in the URL or pipe the file in via STDIN. To pipe a Dockerfile from STDIN:
> 
> ```
> $ docker build - < Dockerfile
> ```

Kubectl seems to use the same one... ish (they use it as a value for the `-f` flag)
```
  # Apply the JSON passed into stdin to a pod.
  cat pod.json | kubectl apply -f -
```